### DISCUSSION TITLE
AM25 Rollerblind motors laggy/unresponsive.

### DISCUSSION THREAD

**Discussion by zegelin** [Koenkk/zigbee2mqtt#8795] [Devices]
Hi!

I've got a bunch of [AM25 roller blind motors](https://www.zigbee2mqtt.io/devices/AM25.html). These support both Zigbee and RF control.  The zigbee2mqtt docs have been updated recently and now label these as [TuYa TS0601_cover](https://www.zigbee2mqtt.io/devices/TS0601_cover.html) devices.

I've been trying on-and-off for the past few months to get the Zigbee control of these motors working reliably. I thought I'd post here and explain what I've tried with the hopes that other who have these motors have some advice (or similar experiences).

Initially I was running OpenHAB paired with a Nortek HUSBZB-1 coordinator (Ember-based). I moved away from OpenHAB for various reasons (mainly their choice not to support TuYa custom attributes, such as battery %, for Zigbee devices) over to zigbee2mqtt. OpenHAB wasn't able to reliably control these motors. They would randomly stop responding. The only fix was to power-cycle the motors.

For the switch to zigbee2mqtt I bought a zzh! stick (CC2652R-based).  I was hoping that different software + coordinator hardware would resolve whatever problem was causing the motors to stop responding (especially since they're listed on z2m's _Supported devices_ page). Unfortunately this doesn't seem to be the case. The same issue occurs. Seemingly randomly, after a period of hours to days, the motors will stop responding to commands (open, close, set position, etc) via the UI or MQTT. Only after power-cycling the motors will they become responsive again for a random period of time.

Even if Zigbee isn't working, the motors never stop responding to RF remote commands. Hence I don't think this is a battery/power issue.

When the devices are responding they do so with quite a bit of lag. Sending open/close/stop can take seconds to respond. While the motors are moving the z2m web-ui's position slider jumps all over the place. Sometimes the final position update doesn't reflect the motor position.

Some things I've tried, with no luck:

- Keeping the motors connected to external power (rather than on battery).
- Re-pairing all the motors
- Trying different Zigbee channels

My latest experiment has been to take a spare RPi and install on it a fresh copy of z2m. I connected to the Pi my zzh! stick and configured z2m to use a different Channel, PAN and Network Key than my original network. I have then paired a single motor to this network. In addition, I've setup my old HUSBZB-1 as a sniffer (using [ZigBeeSniffer.jar](https://github.com/zsmartsystems/com.zsmartsystems.zigbee.sniffer)), and I have the packets going into Wireshark + to pcap files on disk.

I only set this up a few hours ago, and have yet to see the motor become unresponsive. However, even with this super simplified setup I still experience lag when controlling the motor. For example, clicking _OPEN_ in the UI works quite fast, but clicking _CLOSE_ while the motor is opening does nothing until the motor has fully opened and finished sending position updates. _STOP_ is unreliable at best. The _position_ slider still jumps around all over the place, and sometimes never stops at the right position.

The sniffer is working too. I've configured WS with my network key, and I can see decrypted packets in the Wireshark window. When the motor is moving I see a large number of _Report Attribute_ packets.

Even though the PAN and Network Key is different I do see a number _Rejoin Request_ packets. I assume these are from the other blind motors?

My hope is that this is a software or configuration issue, and not a problem with the hardware/firmware of the motors themselves.

Any help would be appreciated.

z2m version: 1.21.0
Coordinator:  `'{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20210120,"transportrev":2},"type":"zStack3x0"}'`

**Comment by zegelin** [comment]
Attached is a screen recording of me interacting with the motor controls in the Z2M UI.

I click _OPEN_. The motor starts moving. You can see the position updates jump around all over the place.
After a few seconds I click _STOP_. This has no effect. The motor keeps moving.
Eventually the motor reaches its open stop-point. Yet the position slider shows 89%.
Clicking _STOP_ again causes the position to update to 100%.
I next click _CLOSE_. This has no effect.
I click _CLOSE_ again. The motor starts moving. The position slider is glitchy again and eventually settles on 3%.
I click _STOP_ and _CLOSE_ multiple times. Eventually the position slider updates to 0%.

https://user-images.githubusercontent.com/19296634/133885693-ed5274fd-ca54-4648-810d-1480f4e4518a.mov

**Reply by zegelin** [reply]
I tried something new this morning.

First I powered off every other blind motor in the house. The Zigbee traffic in Wireshark decreased (even though the other motors should be on the old channel & PAN).

I was also filtering packets in Wireshark to just `zigbee_nwk`. I removed this filter and now I can see the lower-layer data too.

With only a single blind motor powered on and paired with the coordinator I see constant traffic at the `IEEE 802.15.4` layer.
The motor (addr 0xa525) sends a _Data Request_ packet every ~1 second, to which the coordinator _Ack_s:

![packets](https://user-images.githubusercontent.com/19296634/133914121-41c5d5f1-191a-4ee1-8499-92a3934d8249.PNG)

I wasn't sure if this is standard behavior, so I paired an IKEA TRADFRI led bulb. With the motor powered off (i.e, only the bulb is powered on), I see minimal traffic. There's the occasional _Link Status_ ping-pong between the bulb and the coordinator. But not a constant stream of _Data Request_ packets like with the motor.

It seems like these motors spam the network or are in some mis-configured state.  Thoughts?

**Comment by zegelin** [comment]
Reading the docs a bit more I realised that there's two devices listed as _AM25_.

The motors I have are detected by z2m as the the [ORVIBO AM25](https://www.zigbee2mqtt.io/devices/AM25.html), not the [A-OK AM25/TS0601_cover](https://www.zigbee2mqtt.io/devices/TS0601_cover.html).

Confusingly, these motors have a physical sticker identifying them as "A-OK Low Power Tubular Motor, Model: AM25-1.2/30-ES-EZ".

**Comment by zegelin** [comment]
Another update :)

I updated the firmware of my ZZH! to the latest Z-Stack_3.x.0 coordinator firmware for the CC2652R: 20210708

At first I thought this had resolved the issues, as motor position updates were reflected correctly and commands such as CLOSE followed by STOP were responsive.

On deeper inspection this isn't the case. While I had the ZZH! unplugged, the AM25 motor connected directly with the IKEA TRADFRI bulb as its preferred route to the coordinator. Once the ZZH! was powered on again, the network topology shows AM25 -> TRADFRI -> coordinator.

![Capture](https://user-images.githubusercontent.com/19296634/133916460-c1ee37de-db2a-4710-869f-7d9c2560c4c9.PNG)

Wireshark also shows that the continuous stream of _Data Request_ and _Ack_ packets are occurring between the AM25 and the TRADFRI bulb.

Unplugging the bulb forces the AM25 to reroute directly to the coordinator. Once this happens the performance is abysmal again. The position updates come in out-of-order and _STOP_ rarely works.

**Reply by shaundekok** [reply]
Hi, did you ever get this working reliably? I have an issue that the blinds state stays open even once it is closed. After readding it back to the network it updates and works for a say or so. It does not prevent the blind from opening in the morning or closing in the afternoon with an automation.

**Reply by 1006-development** [reply]
@zegelin, I'm having the same issue, they stop working randomly but the RF remote ofc. work, a friend of mine bought the same ones and he bought a zigbee bridge specially for them and if I remember correctly he hasn't had any issues like these (but I didn't want to buy the bridge as it beats the purposes of Home Assistant and just one "bridge" for everything,... Would be nice to get these working with HA.

**Comment by michaelsidenius** [comment]
Hi. just received the same roller (made by A-OK originally I guess: [https://www.a-okmotor.com/roller-blinds-motor/57730802.html]). 

May I ask how you pair with z2m? Cant really find any docs on that. I can also do WS sniffing if necessary.

**Reply by guinnberg** [reply]
I'm having the same issue... It's already driving me crazy! Did you have any luck?

**Comment by michaelsidenius** [comment]
I have not been able to pair yet, unfortunately.

**Reply by guinnberg** [reply]
After some research, I think you won't be able, at least if you have the same model I do (A-OK AM25C).

I contacted straight away with A-OK motors and they confirmed this type of motor only works with the ALLONE PLUS hub from Orvibo, which it's only works with RF and IR.

I haven't researched more, but with that hub, they claim you can use Siri, Alexa and Google Assistant... I'll try to get a refund but if I can't I might end up buying one of those and see how can I integrate it with my ecosystem

**Reply by michaelsidenius** [reply]
I actually got this to pair by pressing the button (only one) down for 5 sec two times and then the LED started blining green. It took me a couple of tries though. Read the method here: https://www.zigbee2mqtt.io/devices/TS0601_cover.html, but it is more like this https://www.zigbee2mqtt.io/devices/AM25.html as zegelin writes.

I can make it run a little in both directions, but the position is not really working.
